site/content/features/gatherers.html

```diff
@@ -0,0 +1,395 @@
+        <li>The <code>combiner</code> yields a function that concatenates two lists, specifically <code>(a, b) -&gt; { a.addAll(b); return a; } </code> </li>
+        <li>The <code>finisher</code> does nothing (which is the most common case)</li>
+      </ul>
+      <p>The static <code>Collectors.groupingBy</code> method yields a more interesting collector. It produces a map from keys to lists of elements with the same key. In its simplest form, you pass a “classifier” function that extracts the map keys from the elements.</p>
+      <p>I like to use the stream <code>Locale.availableLocales()</code> as an example of a useful data stream. A locale describes the language and other preferences of a place. The following call to <code>collect</code> yields a map from country code strings to sets of locales in that country. </p>
+<div class='sandbox'><pre>void main() {
+    Map&lt;String, List&lt;Locale&gt;&gt; countryToLocales = Locale.availableLocales().collect(
+        Collectors.groupingBy(Locale::getCountry));
+    List&lt;Locale&gt; swissLocales = countryToLocales.get("CH");
+        // Yields locales [de_CH, fr_CH, it_CH, rm_CH, ...]
+    println(swissLocales);
+}
+</pre></div>
+      <p>As you can see, the JDK supports a multitude of locales used in Switzerland, including German, French, Italian, and Romansh, all collected in a list.</p>
+      <p>Note the classifier function <code>Locale::getCountry</code> that extracts the map key, i.e. the country code string such as <code>"CH"</code> or <code>"US"</code>.</p>
+      <p>This use of collectors is straightforward. It gets more opaque when you use “downstream collectors” that apply another collection step to the collected map values. Consider:</p>
+<div class='sandbox'>    <pre>void main() {
+    Map&lt;String, Map&lt;String, List&lt;Locale&gt;&gt;&gt; countryAndLanguageToLocale =
+        Locale.availableLocales().collect(
+            groupingBy(Locale::getCountry,
+                groupingBy(Locale::getLanguage)));
+    println(countryAndLanguageToLocale.get("IN").get("hi"));
+}
+</pre></div>
+      <p>Now you have a map of maps. For example, <code>countryAndLanguageToLocale.get("IN").get("hi")</code> is a list of the Hindi locales in India. (There are several variants.)</p>
+      <h2>The Gatherer Interface</h2>
+      <p>JEP 485 introduces a <code>gather</code> method that is analogous to the <code>collect</code> method, but it is an intermediate operation, yielding a stream, not a final result. The method has an argument of type <code>Gatherer</code>, an interface that is similar to the <code>Collector</code> interface. A gatherer has a generic “intermediate state” object instead of the “result collection” of a collector.  It too has four methods yielding function objects. </p>
+      <pre>Supplier&lt;A&gt; initializer()
```

Comment:
My brain was a bit confused by this formatting: I typically expect inline comments **before** the subject. Now that we have nice HTML: What about formatting this a bullet list?

![image](https://github.com/user-attachments/assets/59ef78e7-d6ca-4c77-8687-a70a393cebb3)

Reply:
Using dl instead